### ISSUE TITLE
The --only-deps flag can sometimes install too much

### ISSUE THREAD

**Issue by johanneskoester** [QuantStack/mamba#211]
When specifying 
```
mamba create -c conda-forge -c bioconda -n snakemake snakemake-minimal snakemake --only-deps
```
Mamba wants to install snakemake-minimal as well (because it is a dependency of the snakemake package). However, correct behavior would be not neither install snakemake nor snakemake-minimal. Conda does this correctly.

**Comment by wolfv** [comment]
thanks for this bug report -- and the others!

**Comment by wolfv** [comment]
I have a fix in #256 

However this works slightly different from conda. It will hard ignore all specs that are in the current command when using --only-deps.

Conda has some logic to only remove the leaf packages (which is the "bug" you encountered). To be honest we were using conda's logic here and I agree it seems flawed.
The usecase they are saying is that you can use another package to constrain the leaf package, e.g. 

`conda create -n bla python=2 flask --only-deps`

would use python=2 to constrain the flask package. However, I think it's better to do these kind of things in a two step process or have another flag for constraining packages.

Here is the conda-source for the `only-deps` feature. Judging by the comment conda should also install snakemake-minimal:

https://github.com/conda/conda/blob/2cdd7ab052c2b96c088137f52dc40c613e99ce33/conda/core/solve.py#L866-L894

**Comment by wolfv** [comment]
anyways this should be fixed with #256 
however we still need to add back a way to specify constraints / pins when creating a new environment.
